### DISCUSSION TITLE
cloning project together with eclipse-theia/theia

### DISCUSSION THREAD

**Discussion by RuiCovid** [eclipse-theia/theia#8647] [Q&A] [answered]
Hi 
I cloned eclipse-theia/theia and then ITLingo-studio (for theia) both at the same level in same folder.
I get the error in the image. Why?

![erro8](https://user-images.githubusercontent.com/63452111/96339458-b8fd3580-108c-11eb-903b-fe504a5ea573.jpg)

**Comment by vince-fugnitto** [comment]
@RuiCovid I’m not familiar with `ITLingo-Studio` can you perhaps share additional information and your use-case?

If you are looking into creating your own application then the goal should not be to fork and modify this repo but actually compose an application as described on the website: https://theia-ide.org/docs/composing_applications.

**Comment by RuiCovid** [comment]
Thanks.
First I need to run theia in windows 10.
I will uninstall everything ...
how to do it and how to make it work?
first I had node 12.19.0
(error:

TRACKER : error TRK0005: Failed to locate: "CL.exe". ... )
now
10.15.3
(error: error @theia/monorepo@0.0.0: The engine "node" is incompatible with
this module. Expected version ">=12.14.1 <13". Got "10.15.3)

On Sun, Oct 18, 2020 at 2:28 AM Vincent Fugnitto <notifications@github.com>
wrote:

> @RuiCovid <https://github.com/RuiCovid> I’m not familiar with
> ITLingo-Studio can you perhaps share additional information and your
> use-case?
>
> If you are looking into creating your own application then the goal should
> not be to fork and modify this repo but actually compose an application as
> described on the website:
> https://theia-ide.org/docs/composing_applications.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse-theia/theia/discussions/8647#discussioncomment-102885>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APEDHTYWDOQ4QA46FGDV6F3SLJADPANCNFSM4SUMUKGA>
> .
>


-- 

---------------
Cumprimentos, Obrigado
Rui

**Reply by vince-fugnitto** [reply]
> error: error @theia/monorepo@0.0.0: The engine "node" is incompatible with this module. Expected version ">=12.14.1 <13". Got "10.15.3)
> […](#)

The error states that the node version used (10.15.3) is incompatible, it needs to be a version which satisfies the range.

**Comment by RuiCovid** [comment]
Yes.
But I had 12.19 and got another error... it doesn't make sence...

A domingo, 18/10/2020, 14:10, Vincent Fugnitto <notifications@github.com>
escreveu:

**Comment by RuiCovid** [comment]
How do I uninstall theia, scoop, nvm/npm and yarn to install/clone
everything again? Following the right steps.
Thanks

A domingo, 18/10/2020, 15:54, Rui Carvalho <ruimigcar@gmail.com> escreveu:

**Comment by vince-fugnitto** [comment]
@RuiCovid it is difficult to help without additional information regarding your project structure and use-case.
If you're attempting to build the [theia repository](https://github.com/eclipse-theia/theia) please be sure to install all [prerequisites](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites).

As I previously mentioned, if you're attempting to clone this repo, modify it or include in a separate project this is not the correct procedure. If you're attempting to build your own application please following the guide on [building your own application](https://theia-ide.org/docs/composing_applications/). The `theia` repository is for development of main `theia` extensions and should not be used as your own custom product. Instead, you should compose an application as described in the guide. The title "cloning project together with eclipse-theia/theia" suggests to me that you are trying to somehow use this repo as is for your own project which is incorrect.

**Comment by RuiCovid** [comment]
I'm new to theia.
After cloning theia I make yarn because I think it is needed. Is this true?

A domingo, 18/10/2020, 21:38, Vincent Fugnitto <notifications@github.com>
escreveu:

**Comment by RuiCovid** [comment]
Then I clone to use and modify: itlingo-studio.
I put both in the same folder (same level) or itlingo-studio inside
eclipse-theia solder?

A domingo, 18/10/2020, 21:42, Rui Carvalho <ruimigcar@gmail.com> escreveu:

**Comment by vince-fugnitto** [comment]
@RuiCovid I believe I mentioned twice that there are specific prerequisites, and build instructions when attempting to build the application, and not to modify the project by including it in a separate project.

> I put both in the same folder (same level) or itlingo-studio inside eclipse-theia folder?

I'm not sure why you'd want to do this or what the use-case is, but it is not advised to clone the project so it can be placed in another project. This is a development repository so we can develop `@theia/` extensions, it is not a product not should downstream developers use the project as is to produce an application.

**Comment by RuiCovid** [comment]
Thank you Vincent.
itlingo-studio is an IDE developed with theia.
It is a web version of a desktop tool to test apps or web.

I'm very sorry for the confusion.
I think I understand you now:
I don't need to clone theia to my desktop;
itlingo-studio can be used alone;

My problem is this:
I clone itlingo-studio, then I navigate to itlingo-studio-theia folder and
have to make "yarn rebuild:browser" but get an error message " 'theia' is
not recognized as an internal or external command"

Due to this situation I started thinking that maybe I would need to have
theia locally.
I will delete my copy of theia.

As far as I know I have the requisites but something must be wrong...
Please help.

On Wed, Oct 21, 2020 at 1:34 PM Vincent Fugnitto <notifications@github.com>
wrote:

> @RuiCovid <https://github.com/RuiCovid> I believe I mentioned twice that
> there are specific prerequisites, and build instructions when attempting to
> build the application, and not to modify the project by including it in a
> separate project.
>
> I put both in the same folder (same level) or itlingo-studio inside
> eclipse-theia folder?
>
> I'm not sure why you'd want to do this or what the use-case is, but it is
> not advised to clone the project so it can be placed in another project.
> This is a development repository so we can develop @theia/ extensions, it
> is not a product not should downstream developers use the project as is to
> produce an application.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse-theia/theia/discussions/8647#discussioncomment-107685>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APEDHT2G7S73JT3V4GYPV33SL3IMBANCNFSM4SUMUKGA>
> .
>


-- 

---------------
Cumprimentos, Obrigado
Rui

**Reply by vince-fugnitto** [reply]
@RuiCovid based on the description, you should first build the `itlingo-studio` project (the command `theia` will be available once built which is used by other commands such as `yarn rebuild:browser`). If the `itlingo-stuido` is using theia then it is definitely not needed to add `theia` in the project like you've done.

I do not know the `itlingo-studio` project, if it is public can you link it? Else I think asking the actual `itlingo-studio` project developers the question would be the best solution.

**Comment by RuiCovid** [comment]
If I go to itlingo-studio I must make yarn. I understand.
I will let you later today or tomorrow.



A quarta, 21/10/2020, 14:14, Vincent Fugnitto <notifications@github.com>
escreveu:

**Comment by RuiCovid** [comment]
Better but:
I got failed to locate CL.exe and nsfw.vcxproj

A quarta, 21/10/2020, 14:17, Rui Carvalho <ruimigcar@gmail.com> escreveu:

**Reply by vince-fugnitto** [reply]
@RuiCovid those errors look unrelated to the framework. Please open issues directly in `itlingo-studio`.

**Comment by RuiCovid** [comment, marked as answer]
Done. Thanks

A quarta, 21/10/2020, 14:49, Vincent Fugnitto <notifications@github.com>
escreveu:

**Comment by RuiCovid** [comment]
Hi
Whats the role of package-lock.json ?

Thanks

A quarta, 21/10/2020, 15:03, Rui Carvalho <ruimigcar@gmail.com> escreveu:

**Reply by vince-fugnitto** [reply]
> Hi Whats the role of package-lock.json ? 

There is no `package-lock.json` present in this repository.
As always, the actual [documentation](https://docs.npmjs.com/configuring-npm/package-lock-json.html) would be the best option to reference.
